crates/re_query/src/archetype_view.rs

```diff
@@ -357,6 +359,15 @@ impl<A: Archetype> ArchetypeView<A> {
             // - If the data isn't the same, the cost of the comparison will be dwarfed by the cost
             //   of the join anyway.
             if self.required_comp().instance_keys == component.instance_keys {
+                // This fast iterator is assumed to match the length of the primary component
+                // We shouldn't hit this if since the store should enforce matched lengths for
+                // non-empty components, and the outer if-guard should keep us from reaching this
```

Comment:
again, I don't think this is true for splats, e.g. `Points3D(positions=[a,b,c,d], colors=[rgba])`

Reply:
Splats are currently split into their own rows. See: https://github.com/rerun-io/rerun/issues/1893 -- adding a TODO linking to that issue though.